api/src/repositories/spatial-repository.test.ts

```diff
@@ -0,0 +1,278 @@
+      }
+    });
+
+    it('should succeed with valid data', async () => {
+      const mockQueryResponse = {
+        rowCount: 1,
+        rows: [
+          {
+            json_build_object: {
+              type: 'FeatureCollection',
+              features: []
+            } as FeatureCollection
+          }
+        ]
+      } as any as Promise<QueryResult<any>>;
+
+      const mockDBConnection = getMockDBConnection({
+        query: async () => {
+          return mockQueryResponse;
+        }
+      });
+
+      const spatialRepository = new SpatialRepository(mockDBConnection);
+
+      const response = await spatialRepository.runSpatialTransformOnSubmissionId(1, 'string');
+
+      expect(response).to.eql({
+        type: 'FeatureCollection',
+        features: []
+      });
+    });
+  });
+
+  describe('insertSubmissionSpatialComponent', () => {
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('should throw an error when insert sql fails', async () => {
+      const mockQueryResponse = { rowCount: 0 } as any as Promise<QueryResult<any>>;
+
+      const mockDBConnection = getMockDBConnection({
+        sql: async () => {
+          return mockQueryResponse;
+        }
+      });
+
+      const spatialRepository = new SpatialRepository(mockDBConnection);
+
+      try {
+        await spatialRepository.insertSubmissionSpatialComponent(1, {} as FeatureCollection);
+        expect.fail();
+      } catch (actualError) {
+        expect((actualError as ApiGeneralError).message).to.equal(
+          'Failed to insert submission spatial component details'
+        );
+      }
+    });
+
+    it('should succeed with valid data', async () => {
+      const mockQueryResponse = { rowCount: 1, rows: [{ submission_spatial_component_id: 1 }] } as any as Promise<
+        QueryResult<any>
+      >;
+
+      const mockDBConnection = getMockDBConnection({
+        sql: async () => {
+          return mockQueryResponse;
+        }
+      });
+
+      const spatialRepository = new SpatialRepository(mockDBConnection);
+
+      const response = await spatialRepository.insertSubmissionSpatialComponent(1, {} as FeatureCollection);
+
+      expect(response.submission_spatial_component_id).to.equal(1);
+    });
+
+    it('should succeed with valid data and append geography to sql statment', async () => {
+      const mockQueryResponse = { rowCount: 1, rows: [{ submission_spatial_component_id: 1 }] } as any as Promise<
+        QueryResult<any>
+      >;
+
+      const mockDBConnection = getMockDBConnection({
+        sql: async () => {
+          return mockQueryResponse;
+        }
+      });
+
+      const generateGeometryCollectionSQLStub = sinon
+        .stub(spatialUtils, 'generateGeometryCollectionSQL')
+        .returns(SQL`valid`);
+
+      const spatialRepository = new SpatialRepository(mockDBConnection);
+
+      const response = await spatialRepository.insertSubmissionSpatialComponent(1, {
+        features: [
```

Comment:
Can you add the `type: 'FeatureCollection'` property? Its the only thing missing that would make this a proper valid feature collection.

Reply:
You may be able to remove the `as FeatureCollection` as a result.